### PR TITLE
Singularity containers with cloud build review

### PR DESCRIPTION
**Pull request by wkharold**
This pull request adds a new tutorial that demonstrates how to use Cloud Build to
* Create a custom build step that constructs Singularity containers
* Use the custom Singularity build step to actually build a simple Singularity container

The tutorial also helps the user create a Compute Instance with Singularity installed that can be used to verity the constructed container(s).

Singularity is a popular alternative container technology that is used heavily in HPC environments.

### CODE REVIEW

File: tutorials/singularity-containers-with-cloud-build/README.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
+---
+title: Building Singularity Containers using Cloud Build
+description: Learn how to use Cloud Build to build Singularity Containers for HPC workloads
+author: wkharold, vsoch
+tags: Cloud Build, Singularity, HPC
+date_published: 2019-01-18
+---
+
+This tutorial shows you how to use [Cloud Build](https://cloud.google.com/cloud-build/) to build [Singularity](https://www.sylabs.io/singularity/) containers. 
+In constrast to [Docker](https://www.docker.com/), the Singularity container binary is designed specifically for HPC workloads.
```
Comment (dghale) [inline tutorials/singularity-containers-with-cloud-build/README.md]: Recommend spelling out High Performance Computing (HPC) on first use.

File: tutorials/singularity-containers-with-cloud-build/README.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+![diagram](overview.png)
+
+[![button](http://gstatic.com/cloudssh/images/open-btn.png)](https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/community&page=editor&tutorial=tutorials/singularity-containers-with-cloud-build/tutorial.md)
+
+## Before you begin
+
+### Install the [Google Cloud SDK](https://cloud.google.com/sdk/install)
+
+The Cloud SDK installs commands and libs for interacting with Cloud Platform from the command line. This
```
Comment (dghale) [inline tutorials/singularity-containers-with-cloud-build/README.md]: Google Cloud Platform

File: tutorials/singularity-containers-with-cloud-build/README.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+tutorial makes extensive use of the ```gcloud``` and ``gsutil``` commands to create and manage Cloud Platform
+resources.
```
Comment (dghale) [inline tutorials/singularity-containers-with-cloud-build/README.md]: missing a backtick before gsutil?

File: tutorials/singularity-containers-with-cloud-build/README.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+```
+
+Then use this command to build the custom build step.
+
+```bash
+gcloud builds submit --config=builder.yaml --substitutions=_SINGULARITY_VERSION="3.0.2" .
+```
+
+At the time of this writing the latest stable version of Singularity is 3.0.2. To use a different, possibly newer, version modify the value of the
+_SINGULARITY_VERSION substitution accordingly. What does this verison coincide with? It should match a [release tag](https://github.com/sylabs/singularity/releases)
```
Comment (dghale) [inline tutorials/singularity-containers-with-cloud-build/README.md]: Need to escape the underscore at beginning with a backslash: \_SINGU...

Comment (dghale) [inline tutorials/singularity-containers-with-cloud-build/README.md]: :s/substitution/substitute

File: tutorials/singularity-containers-with-cloud-build/env.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+ORG=[YOUR_ORG]
+BILLING_ACCOUNT=[YOUR_BILLING_ACCOUNT_NAME]
+PROJECT=[NAME FOR THE PROJECT YOU WILL CREATE]
```
Comment (dghale) [inline tutorials/singularity-containers-with-cloud-build/env.sh]: any reason not to stick with the convention of using underscores between words for the last two?

File: tutorials/singularity-containers-with-cloud-build/README.md (hunk 1 above)
Comment (wkharold) [reply tutorials/singularity-containers-with-cloud-build/README.md]: Done

File: tutorials/singularity-containers-with-cloud-build/README.md (hunk 2 above)
Comment (wkharold) [reply tutorials/singularity-containers-with-cloud-build/README.md]: Done

File: tutorials/singularity-containers-with-cloud-build/README.md (hunk 3 above)
Comment (wkharold) [reply tutorials/singularity-containers-with-cloud-build/README.md]: Done

File: tutorials/singularity-containers-with-cloud-build/README.md (hunk 4 above)
Comment (wkharold) [reply tutorials/singularity-containers-with-cloud-build/README.md]: Done

Comment (wkharold) [reply tutorials/singularity-containers-with-cloud-build/README.md]: Done

File: tutorials/singularity-containers-with-cloud-build/env.sh (hunk 5 above)
Comment (wkharold) [reply tutorials/singularity-containers-with-cloud-build/env.sh]: Nope. Done

**Comment by vsoch** [comment]
Dearest @googlebot, I can confirm that I am ok with my commits being contributed to this project!

**Comment by vsoch** [comment]
@dghale I'm fairly excited about Google Cloud Build, generally, for different environments to issue cloud builds. I'm working on some tooling for singularity registry client (with great teamwork with @wkharold along the way!) and it's giving me a lot of ideas for how this would fit even into powering the back end of Singularity Hub. Hugely grateful to have contributed here, and to see this move through!

**Comment by wardharold** [comment]
I confirm that I approve of my commits being contributed to the project.

**Comment by ToddKopriva** [comment]
I have added the cla:yes label because author Vanessa Sochat (vsoch) has indicated her explicit consent in the comments above.

**Comment by ToddKopriva** [comment]
published:
https://cloud.google.com/community/tutorials/singularity-containers-with-cloud-build

Thanks, Ward and Vanessa.

**Comment by vsoch** [comment]
Thanks as well @ToddKopriva and @googlebot :)

**Pull request merged**